src/components/model/Models.svelte

```diff
@@ -0,0 +1,373 @@
+            }}
+          >
+            Edit details and associations
+          </button>
+        </div>
+      {:else}
+        <form on:submit|preventDefault={submitForm}>
+          <AlertError class="m-2" error={$createModelError} />
+
+          <fieldset>
+            <label for="name">Name</label>
+            <input
+              bind:value={name}
+              autocomplete="off"
+              class="st-input w-100"
+              name="name"
+              required
+              use:permissionHandler={{
+                hasPermission: hasCreateModelPermission,
+                permissionError: createModelPermissionError,
+              }}
+            />
+          </fieldset>
+
+          <fieldset>
+            <label for="version">Version</label>
+            <input
+              bind:value={version}
+              autocomplete="off"
+              class="st-input w-100"
+              name="version"
+              placeholder="0.0.0"
+              required
+              use:permissionHandler={{
+                hasPermission: hasCreateModelPermission,
+                permissionError: createModelPermissionError,
+              }}
+            />
+          </fieldset>
+
+          <fieldset>
+            <label for="description">Description</label>
+            <textarea
+              bind:value={description}
+              autocomplete="off"
+              class="st-input w-100"
+              name="description"
+              placeholder="Enter Model Description (optional)"
+            />
+          </fieldset>
+
+          <fieldset>
+            <label for="file">Jar File</label>
+            <input
+              class="w-100"
+              name="file"
+              required
+              type="file"
+              bind:files
+              use:permissionHandler={{
+                hasPermission: hasCreateModelPermission,
+                permissionError: createModelPermissionError,
+              }}
+            />
+          </fieldset>
+
+          <fieldset>
+            <button
+              class="st-button w-100"
+              disabled={createButtonDisabled}
+              type="submit"
+              use:permissionHandler={{
+                hasPermission: hasCreateModelPermission,
+                permissionError: createModelPermissionError,
+              }}
+            >
+              {$creatingModel ? 'Creating...' : 'Create'}
+            </button>
+          </fieldset>
+        </form>
+      {/if}
+    </svelte:fragment>
+  </Panel>
+
+  <Panel>
+    <svelte:fragment slot="header">
+      <SectionTitle>
+        <BarChartIcon />
+        Models
+      </SectionTitle>
+    </svelte:fragment>
+
+    <svelte:fragment slot="body">
+      {#if $models.length}
+        <SingleActionDataGrid
+          {columnDefs}
+          hasDeletePermission={hasDeleteModelPermission}
+          itemDisplayText="Model"
+          items={$models}
+          {user}
+          selectedItemId={selectedModel?.id}
```

Comment:
If I don't have a model selected I can't seem to click on the edit icon button. Need to first click on the row and then click on the edit button to go into the editor.